ding/reward_model/curiosity_reward_model.py

```diff
@@ -0,0 +1,201 @@
+from typing import Union, Tuple
+from easydict import EasyDict
+
+import random
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import torch.nn.functional as F
+
+from ding.utils import SequenceType, REWARD_MODEL_REGISTRY
+from ding.model import FCEncoder, ConvEncoder
+from ding.torch_utils import one_hot
+from .base_reward_model import BaseRewardModel
+
+
+def collect_states(iterator: list) -> Tuple[list, list, list]:
+    states = []
+    next_states = []
+    actions = []
+    for item in iterator:
+        state = item['obs']
+        next_state = item['next_obs']
+        action = item['action']
+        states.append(state)
+        next_states.append(next_state)
+        actions.append(action)
+    return states, next_states, actions
+
+
+class ICMNetwork(nn.Module):
+    r"""
+    Intrinsic Curiosity Model (ICM Module)
+    Implementation of:
+    [1] Curiosity-driven Exploration by Self-supervised Prediction
+    Pathak, Agrawal, Efros, and Darrell - UC Berkeley - ICML 2017.
+    https://arxiv.org/pdf/1705.05363.pdf
+
+    1) Embedding observations into a latent space
+    2) Predicting the action logit given two consecutive embedded observations
+    3) Predicting the next embedded obs, given the embeded former observation and action
+    """
+
+    def __init__(self, obs_shape: Union[int, SequenceType], hidden_size_list: SequenceType, action_shape: int) -> None:
+        super(ICMNetwork, self).__init__()
+        if isinstance(obs_shape, int) or len(obs_shape) == 1:
+            self.feature = FCEncoder(obs_shape, hidden_size_list)
+        elif len(obs_shape) == 3:
+            self.feature = ConvEncoder(obs_shape, hidden_size_list)
+        else:
+            raise KeyError(
+                "not support obs_shape for pre-defined encoder: {}, please customize your own ICM model".
+                format(obs_shape)
+            )
+        self.action_shape = action_shape
+        feature_output = hidden_size_list[-1]
+        self.inverse_net = nn.Sequential(nn.Linear(feature_output * 2, 512), nn.ReLU(), nn.Linear(512, action_shape))
+        self.residual = [
+            nn.Sequential(
+                nn.Linear(action_shape + 512, 512),
+                nn.LeakyReLU(),
+                nn.Linear(512, 512),
+            ) for _ in range(8)
+        ]
+        self.forward_net_1 = nn.Sequential(nn.Linear(action_shape + feature_output, 512), nn.LeakyReLU())
+        self.forward_net_2 = nn.Sequential(nn.Linear(action_shape + 512, feature_output), )
```

Comment:
you can just use `self.forward_net_2 = nn.Linear(action_shape + 512, feature_output)` here